### PR TITLE
[15.0][MIG] mail_improved_tracking_value: Migration to 15.0 

### PR DESCRIPTION
**Pull request by duyanhk15**
ref: BSIDM-644

### CODE REVIEW

**Review by cyrilmanuel — APPROVED** [review]
LGTM

**Review by gurneyalex — APPROVED** [review]

**Review by Ricardoalso — APPROVED** [review]

File: mail_improved_tracking_value/__manifest__.py [hunk 1]
```diff
@@ -1,9 +1,9 @@
-# Copyright 2018 Camptocamp SA
+# Copyright 2018-2022 Camptocamp SA
```
Comment (simahawk) [inline mail_improved_tracking_value/__manifest__.py]: pls rollback these changes in the copyright. The first year means "from".

**Comment by square-ava** [comment]
@duyanhk15 Can you do the changes in review ?

**Comment by sonhd91** [comment]
> @duyanhk15 Can you do the changes in review ?

Hello, I just help a bit, please check again ?

**Comment by Leonidas-VII** [comment]
Hi, I have an idea here. Would that be better if we set the value of the "For" field with the "id" of the mail message instead of "False"?
Even if it is False, I still can click and access the mail message. It doesn't seem to be very good in a functional way.

image

image

**Comment by ajaniszewska-dev** [comment]
@OCA/social-maintainers could we merge this one?

**Comment by Saran440** [comment]
I test change value in line it's error.

Step to error
1. install module 'mail_improved_tracking_value' and 'account'
2. Go to 'Invoicing > Accounting > Journal Items'
3. Select 1 line and edit Due Date > Save
Selection_015
5. Go to Settings > Technical > Improved Tracking Values

```RPC_ERROR
Odoo Server Error
Traceback (most recent call last):
  File "/opt/odoo/custom/src/odoo/odoo/api.py", line 886, in get
    return field_cache[record._ids[0]]
KeyError: 5

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/odoo/custom/src/odoo/odoo/fields.py", line 1082, in __get__
    value = env.cache.get(record, self)
  File "/opt/odoo/custom/src/odoo/odoo/api.py", line 889, in get
    raise CacheMiss(record, field)
odoo.exceptions.CacheMiss: 'mail.tracking.value(5,).old_value_formatted'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/odoo/custom/src/odoo/odoo/api.py", line 886, in get
    return field_cache[record._ids[0]]
KeyError: 5

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/odoo/custom/src/odoo/odoo/fields.py", line 1135, in __get__
    value = env.cache.get(record, self)
  File "/opt/odoo/custom/src/odoo/odoo/api.py", line 889, in get
    raise CacheMiss(record, field)
odoo.exceptions.CacheMiss: 'mail.tracking.value(5,).old_value_formatted'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/odoo/custom/src/odoo/odoo/addons/base/models/ir_http.py", line 237, in _dispatch
    result = request.dispatch()
  File "/opt/odoo/custom/src/odoo/odoo/http.py", line 698, in dispatch
    result = self._call_function(**self.params)
  File "/opt/odoo/custom/src/odoo/odoo/http.py", line 368, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/opt/odoo/custom/src/odoo/odoo/service/model.py", line 94, in wrapper
    return f(dbname, *args, **kwargs)
  File "/opt/odoo/custom/src/odoo/odoo/http.py", line 357, in checked_call
    result = self.endpoint(*a, **kw)
  File "/opt/odoo/custom/src/odoo/odoo/http.py", line 921, in __call__
    return self.method(*args, **kw)
  File "/opt/odoo/custom/src/odoo/odoo/http.py", line 546, in response_wrap
    response = f(*args, **kw)
  File "/opt/odoo/auto/addons/web/controllers/main.py", line 1282, in search_read
    return self.do_search_read(model, fields, offset, limit, domain, sort)
  File "/opt/odoo/auto/addons/web/controllers/main.py", line 1301, in do_search_read
    return Model.web_search_read(domain, fields, offset=offset, limit=limit, order=sort)
  File "/opt/odoo/auto/addons/web/models/models.py", line 62, in web_search_read
    records = self.search_read(domain, fields, offset=offset, limit=limit, order=order)
  File "/opt/odoo/custom/src/odoo/odoo/models.py", line 5064, in search_read
    result = records.read(fields, **read_kwargs)
  File "/opt/odoo/custom/src/odoo/odoo/models.py", line 3232, in read
    return self._read_format(fnames=fields, load=load)
  File "/opt/odoo/custom/src/odoo/odoo/models.py", line 3252, in _read_format
    vals[name] = convert(record[name], record, use_name_get)
  File "/opt/odoo/custom/src/odoo/odoo/models.py", line 5902, in __getitem__
    return self._fields[key].__get__(self, type(self))
  File "/opt/odoo/custom/src/odoo/odoo/fields.py", line 1138, in __get__
    raise ValueError("Compute method failed to assign %s.%s" % (record, self.name))
Exception

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/odoo/custom/src/odoo/odoo/http.py", line 654, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/opt/odoo/custom/src/odoo/odoo/http.py", line 301, in _handle_exception
    raise exception.with_traceback(None) from new_cause
ValueError: Compute method failed to assign mail.tracking.value(5,).old_value_formatted
```


and can you add new folder `readme`?

**Comment by sonhd91** [comment]
> I test change value in line it's error.
> 
> Step to error
> 
> 1. install module 'mail_improved_tracking_value' and 'account'
> 2. Go to 'Invoicing > Accounting > Journal Items'
> 3. Select 1 line and edit Due Date > Save
>    Selection_015
> 4. Go to Settings > Technical > Improved Tracking Values
> 
> ```
> Odoo Server Error
> Traceback (most recent call last):
>   File "/opt/odoo/custom/src/odoo/odoo/api.py", line 886, in get
>     return field_cache[record._ids[0]]
> KeyError: 5
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/opt/odoo/custom/src/odoo/odoo/fields.py", line 1082, in __get__
>     value = env.cache.get(record, self)
>   File "/opt/odoo/custom/src/odoo/odoo/api.py", line 889, in get
>     raise CacheMiss(record, field)
> odoo.exceptions.CacheMiss: 'mail.tracking.value(5,).old_value_formatted'
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/opt/odoo/custom/src/odoo/odoo/api.py", line 886, in get
>     return field_cache[record._ids[0]]
> KeyError: 5
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/opt/odoo/custom/src/odoo/odoo/fields.py", line 1135, in __get__
>     value = env.cache.get(record, self)
>   File "/opt/odoo/custom/src/odoo/odoo/api.py", line 889, in get
>     raise CacheMiss(record, field)
> odoo.exceptions.CacheMiss: 'mail.tracking.value(5,).old_value_formatted'
> 
> The above exception was the direct cause of the following exception:
> 
> Traceback (most recent call last):
>   File "/opt/odoo/custom/src/odoo/odoo/addons/base/models/ir_http.py", line 237, in _dispatch
>     result = request.dispatch()
>   File "/opt/odoo/custom/src/odoo/odoo/http.py", line 698, in dispatch
>     result = self._call_function(**self.params)
>   File "/opt/odoo/custom/src/odoo/odoo/http.py", line 368, in _call_function
>     return checked_call(self.db, *args, **kwargs)
>   File "/opt/odoo/custom/src/odoo/odoo/service/model.py", line 94, in wrapper
>     return f(dbname, *args, **kwargs)
>   File "/opt/odoo/custom/src/odoo/odoo/http.py", line 357, in checked_call
>     result = self.endpoint(*a, **kw)
>   File "/opt/odoo/custom/src/odoo/odoo/http.py", line 921, in __call__
>     return self.method(*args, **kw)
>   File "/opt/odoo/custom/src/odoo/odoo/http.py", line 546, in response_wrap
>     response = f(*args, **kw)
>   File "/opt/odoo/auto/addons/web/controllers/main.py", line 1282, in search_read
>     return self.do_search_read(model, fields, offset, limit, domain, sort)
>   File "/opt/odoo/auto/addons/web/controllers/main.py", line 1301, in do_search_read
>     return Model.web_search_read(domain, fields, offset=offset, limit=limit, order=sort)
>   File "/opt/odoo/auto/addons/web/models/models.py", line 62, in web_search_read
>     records = self.search_read(domain, fields, offset=offset, limit=limit, order=order)
>   File "/opt/odoo/custom/src/odoo/odoo/models.py", line 5064, in search_read
>     result = records.read(fields, **read_kwargs)
>   File "/opt/odoo/custom/src/odoo/odoo/models.py", line 3232, in read
>     return self._read_format(fnames=fields, load=load)
>   File "/opt/odoo/custom/src/odoo/odoo/models.py", line 3252, in _read_format
>     vals[name] = convert(record[name], record, use_name_get)
>   File "/opt/odoo/custom/src/odoo/odoo/models.py", line 5902, in __getitem__
>     return self._fields[key].__get__(self, type(self))
>   File "/opt/odoo/custom/src/odoo/odoo/fields.py", line 1138, in __get__
>     raise ValueError("Compute method failed to assign %s.%s" % (record, self.name))
> Exception
> 
> The above exception was the direct cause of the following exception:
> 
> Traceback (most recent call last):
>   File "/opt/odoo/custom/src/odoo/odoo/http.py", line 654, in _handle_exception
>     return super(JsonRequest, self)._handle_exception(exception)
>   File "/opt/odoo/custom/src/odoo/odoo/http.py", line 301, in _handle_exception
>     raise exception.with_traceback(None) from new_cause
> ValueError: Compute method failed to assign mail.tracking.value(5,).old_value_formatted
> ```
> 
> and can you add new folder `readme`?

I just updated the branch and adding `readme` folder, please check again

**Comment by simahawk** [comment]
@Saran440 good for you?

**Comment by Saran440** [comment]
I still found error

Step to test:
1. Go to 'Invoicing > Accounting > Journal Items'
2. Select 1 line and edit Due Date > Save
3. Go to Settings > Technical > Improved Tracking Values -> **It's work**
4. Create new vendor bills with 1 line > confirm
5. Go to Settings > Technical > Improved Tracking Values -> **Same Error**

**Comment by sonhd91** [comment]
> I still found error
> 
> Step to test:
> 
> 1. Go to 'Invoicing > Accounting > Journal Items'
> 2. Select 1 line and edit Due Date > Save
> 3. Go to Settings > Technical > Improved Tracking Values -> **It's work**
> 4. Create new vendor bills with 1 line > confirm
> 5. Go to Settings > Technical > Improved Tracking Values -> **Same Error**

Sorry for delay, please checked it again

**Pull request merged**